### PR TITLE
Evaluator Server Update

### PR DESCRIPTION
**Pull request by kiroco12**
Resolved compilation problems caused by the dependencies update. This PR does not solve compilation problems in the test module, this will be added in a future PR.

This evaluator resolves dependencies, compiles and executes code using a REST API. This request code has been used to test the evaluator using postman:

`{
  "code": "{import cats.effect._\nimport cats.syntax.all._\nimport fetch._\nimport cats.data.NonEmptyList\n\ntrait DataSource[F[_], Identity, Result]{\n  def data: Data[Identity, Result]\n\n  def CF: Concurrent[F]\n\n  def fetch(id: Identity): F[Option[Result]]\n\n  def batch(ids: NonEmptyList[Identity]): F[Map[Identity, Result]]\n}\ntype UserId = Int\ncase class User(id: UserId, username: String)\n\ndef latency[F[_] : Concurrent](msg: String): F[Unit] = for {\n  _ <- Sync[F].delay(println(s\"--> [${Thread.currentThread.getId}] $msg\"))\n_ <- Sync[F].delay(Thread.sleep(100))\n  _ <- Sync[F].delay(println(s\"<-- [${Thread.currentThread.getId}] $msg\"))\n} yield ()}",
  "resolvers": [
    "https://oss.sonatype.org/content/repositories/snapshots",
    "https://oss.sonatype.org/content/repositories/public"
  ],
  "dependencies": [
    {
      "groupId": "com.47deg",
      "version": "1.2.0",
      "artifactId": "fetch_2.11"
    }
  ]
}`

Remaining parts to finish the evaluator update are the solution of compilation problems in the server/test module and the creation of a new http client for the client module.

### CODE REVIEW

File: project/ProjectPlugin.scala [hunk 1]
```diff
@@ -81,9 +82,12 @@ object ProjectPlugin extends AutoPlugin {
...
         %("config"),
         %%("jwt-core", V.jwtCore),
+        %%("coursier", V.coursier),
+        %%("coursier-cache", V.coursier),
+        "io.get-coursier"            %% "coursier-cats-interop" % V.coursier,
         %%("scalatest", V.scalatest) % "test"
-      ),
-      addSbtPlugin("io.get-coursier" % "sbt-coursier" % "2.0.0-RC3-2")
+      ) //,
+      //addSbtPlugin("io.get-coursier" % "sbt-coursier" % V.coursier)
```
Comment (calvellido) [inline project/ProjectPlugin.scala]: :scream:

File: server/src/test/scala/org/scalaexercises/evaluator/EvaluatorSpec.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 package org.scalaexercises.evaluator
 
-import monix.execution.Scheduler
+//import monix.execution.Scheduler
```
Comment (calvellido) [inline server/src/test/scala/org/scalaexercises/evaluator/EvaluatorSpec.scala]: Could we remove all the unnecessary commented out code entries?

File: server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala [hunk 3]
```diff
@@ -17,7 +17,7 @@ import org.scalatest._
 import pdi.jwt.{Jwt, JwtAlgorithm}
 import scodec.bits.ByteVector
 
-import scalaz.stream.Process.emit
+//import scalaz.stream.Process.emit
```
Comment (calvellido) [inline server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala]: Could we remove all the unnecessary commented out code entries?

File: server/src/main/scala/org/scalaexercises/evaluator/services.scala [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
   import EvalResponse.messages._
-  import codecs._
-  import io.circe.generic.auto._
 
-  private val logger = getLogger
-
-  implicit val scheduler: Scheduler = Scheduler.io("scala-evaluator")
-
-  val evaluator = new Evaluator(20 seconds)
+  def evaluator[F[_]: ConcurrentEffect: ContextShift: Timer: Sync] = new Evaluator[F](20 seconds)
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/services.scala]: My concern here is that we would be creating a new evaluator instance for every request

**Pull request merged**